src/partitioner/local/RDFPartitioner.cpp

```diff
@@ -57,19 +61,128 @@ void RDFPartitioner::loadDataSet(string inputFilePath, string outputFilePath, in
         std::istringstream stream(line);
         std::getline(stream, subject_str, splitter);
         stream >> predicate_str >> object_str;
-        std::cout << "subject_str---" << subject_str << std::endl;
-        std::cout << "predicate_str---" << predicate_str << std::endl;
-        std::cout << "object_str---" << object_str << std::endl;
 
 
+
+        long firstVertex = addToStore(&nodes, subject_str);
+
+
+        long relation = addToStore(&predicates, predicate_str);
+
+        long secondVertex = addToStore(&nodes, object_str);
+
+
+        std::string secondVertexStr = std::to_string(secondVertex);
+
+        addToMap(&relationsMap, firstVertex, relation, secondVertexStr);
+
+
+        std::set<long> vertexSet = graphStorage.find(firstVertex)->second;
+
+        for (auto it = vertexSet.begin(); it != vertexSet.end(); ++it)
+
+            if (vertexSet.empty()) {
+
+                vertexSet = set<long>();
+                vertexSet.insert(secondVertex);
+                edgeCount++;
+                graphStorage.insert({firstVertex, vertexSet});
+            } else if (!vertexSet.empty()) {
+
+                if (vertexSet.insert(secondVertex).second) {
+                    edgeCount++;
+                }
+            }
+
     }
+    writeRelationData();
 }
 
+long RDFPartitioner::addToStore(std::map<string, long> *map, string URI) {
+    long id;
+    id = map->size();
+
+    auto search = map->find(URI);
+    if (search != map->end()) {
+        return search->second;
+    } else {
+
+    }
+
+
+    if (*map == nodes) {
+        nodesTemp.insert({id, URI});
+
+
+    } else if (*map == predicates) {
+        predicatesTemp.insert({id, URI});
+
+    }
+    map->insert({URI, id});
+    return id;
+}
+
+
+void RDFPartitioner::addToMap(std::map<long, std::map<long, std::vector<string> >> *map, long vertex, long relation,
+                              string value) {
+    std::map<long, std::vector<string>> miniMap = map->find(vertex)->second;
+
+
+    if (!miniMap.empty()) {
+
+        std::vector<string> list = miniMap.find(0)->second;
+
+        if (!list.empty()) {
+
+            list.push_back(value);
+        } else if (list.empty()) {
+
+            list = std::vector<string>();
+            list.push_back(value);
+            miniMap.insert({relation, list});
+        }
+
+    } else if (miniMap.empty()) {
```

Comment:
You can replace `else if (miniMap.empty())` with `else`.

Reply:
Yes. While I am testing it went to the else statement but it was not because the case is miniMap.empty(). After fixing that I intended to replace it by else.